src/ucs/type/spinlock.h

```diff
@@ -40,50 +51,91 @@ typedef struct ucs_recursive_spinlock {
 } ucs_recursive_spinlock_t;
 
 
-static ucs_status_t ucs_spinlock_init(ucs_spinlock_t *lock, int flags)
+void ucs_recursive_spinlock_destroy(ucs_recursive_spinlock_t *lock);
+
+/**
+ * Spinlock implementation section
+ */
+static UCS_F_ALWAYS_INLINE void ucs_spinlock_init(ucs_spinlock_t *lock)
 {
-    int ret, lock_flags;
+    lock->lock = UCS_SPINLOCK_FREE;
+}
 
-    if (flags & UCS_SPINLOCK_FLAG_SHARED) {
-        lock_flags = PTHREAD_PROCESS_SHARED;
-    } else {
-        lock_flags = PTHREAD_PROCESS_PRIVATE;
+static UCS_F_ALWAYS_INLINE void ucs_spinlock_destroy(ucs_spinlock_t *lock)
```

Comment:
let's remove this function in next PR, including recursive. We can comment the spinlock does not require cleanup